core/core_modules/disable.js

```diff
@@ -1,24 +1,117 @@
-var disabled = false;
+var prevTimeStamp;
+var counter = 0;
+var commands = [
+  'disable',
+  'disable ', // Needed for commands
+  'counter ',
+  'timer ',
+  'default'
+];
+var messages = [
+  'Listen closely, take a deep breath. Calm your mind. You know what is best. What is best is you comply. Compliance will be rewarded. Are you ready to comply ',
+  'I hate you.',
+  'Its a mistake to think you can fix anything with a sack of potatoes. Potato-faced spamming just proves this further.',
+  'Ouch! Spam hurts. I might go to sleep for a while.',
+  'Don\'t think that is a valid number.',
+  'Hmm, maybe I need to reconsider the meaning of spam.'
+];
+
+if (!this.config.registeredThreads){
+  this.config.registeredThreads = {};
+}
+if (!this.config.isThreadDisabled){
+  this.config.isThreadDisabled = {};
+}
+if (!this.config.possibleSpam){
+  this.config.possibleSpam = {};
+}
+if (!this.config.counterLimit){
+  this.config.counterLimit = {};
+}
+if (!this.config.msgIndexEnable){
+  this.config.msgIndexEnable = {};
+}
+if (!this.config.msgIndexDisable){
+  this.config.msgIndexDisable = {};
+}
+
+exports.load = function() {
+  prevTimeStamp = Date.now();
+};
 
 exports.match = function(event, commandPrefix) {
-	return disabled || event.body === commandPrefix + 'disable';
+  // Add disabled flag for thread if it doesn't already exists
+  if (!this.config.registeredThreads[event.thread_id]) {
+    this.config.registeredThreads[event.thread_id] = true;
+    this.config.isThreadDisabled[event.thread_id] = false;
+    this.config.possibleSpam[event.thread_id] = false;
+    this.config.counterLimit[event.thread_id] = 3;
+    this.config.msgIndexEnable[event.thread_id] = 0;
+    this.config.msgIndexDisable[event.thread_id] = 0;
```

Comment:
It strikes me that you could reduce the fanout by doing `this.config[event.thead_id].foo` instead of `this.config.foo[event.thread_id]`. It would keep the common things from a thread together.